### ISSUE TITLE
Liver Emu server for public testing

### ISSUE THREAD

**Issue by huard** [bird-house/emu#41]
## Description

I think it would be useful to have a public live emu server that can be used for testing. The server would have both anonymous access and known credentials (testuser, testpassword) so we can test clients.

**Comment by cehbrecht** [comment]
Need on the WPS client side testing:
https://github.com/bird-house/birdy/issues/36

**Comment by huard** [comment]
@tomLandry Is this something CRIM could look at ?

**Comment by tomLandry** [comment]
Probably, but on the medium-term track. Meaning probably not this summer.

**Comment by cehbrecht** [comment]
See also: https://github.com/bird-house/bird-house.github.io/issues/24

**Comment by huard** [comment]
We have that now, right ?

**Comment by cehbrecht** [comment]
@huard If you mean the one at DKRZ, it is only a demo installation and not maintained for testing. We probably want to have one Emu which is used for testing only and updated from release to release.

**Comment by huard** [comment]
@tomLandry Is this something CRIM could contribute?

**Comment by tomLandry** [comment]
Trying to wrap my mind on how/if this is useful for the Compute Challenge.

**Comment by huard** [comment]
At the moment Emu has a subset process that complies (or should) with the CWT API. One thing we could do is run the certification script on this process, and eventually implement the other services (aggregate, regrid). 

More generally, we use Emu to test clients. For example, WPS should in principle work with identifiers that include &;-. and unicode characters. There is a process in Emu with such characters and that revealed bugs in PyWPS, birdy and possibly owslib (unconfirmed). We could use Emu to test security and so on. Emu allows us to create dummy processes that can exercise corner cases, instead of the happy path we restrict ourselves to with production servers.

**Comment by tomLandry** [comment]
Cool, thanks. I'm currently surveying issues related to ESGF and CWT that are within reach and useful short term - before creating new ones with help of David Byrns. We still have flexibility in how we present that handful of taks to OGC. Of course I have https://github.com/bird-house/birdy/issues/102 and https://github.com/bird-house/birdy/issues/58 on my radar.

**Comment by cehbrecht** [comment]
> At the moment Emu has a subset process that complies (or should) with the CWT API. One thing we could do is run the certification script on this process, and eventually implement the other services (aggregate, regrid).

@huard @tomLandry 
Do you think it would be useful to have a dedicated WPS service (bird) for the CWT API?

**Comment by huard** [comment]
@cehbrecht Hum. Yes, I suppose you're right. But there is still a need for a live test server IMO.

**Comment by tomLandry** [comment]
I think we need a live test server (or only service?) for CWT API. If you look at OGC Testbed-15 Call For Participation, it says "_The Testbed-15 solution will be tested with platforms provided by NASA, ESA, and ideally the ESGF_." That assumes that once TB-15 is started, whoever get the D124 Deliverable will like us very much if we give a headstart to interoperate with NASA/LLNL stacks and endpoints through the API. For "ESA platforms", that roughly translates to Twitcher in WPS 2.0 REST and enabling the Thematic Exploitation Platform configuration. See here: https://portal.opengeospatial.org/files/?artifact_id=82290#EOPAD

**Comment by huard** [comment]
A new full implementation of the CWT API backend with PyWPS would be a lot of work.

**Comment by tomLandry** [comment]
Yes, mucho work. My intention for the OGC-DOE project (in the associated "ESGF Compute" Engineering Report) is to neatly document how it could be done, to broadly design the solution. If it's compelling, there will be participants of OGC's testbed paid to take this in account in their own implementations. Also, that's the type of work we'd like to do in DACCS.
